cmd/cstor-volume-mgmt/controller/volume-controller/handler.go

```diff
@@ -79,6 +84,23 @@ func (c *CStorVolumeController) syncHandler(
 		}
 		return err
 	}
+	glog.Infof("[Debug] Comparing desired and current cpacity")
+	desiredCapacity, err := resource.ParseQuantity(cStorVolumeGot.Spec.Capacity)
+	if err != nil {
+		pkg_errors.Wrapf(
+			err, "failed to update capacity: failed to parse capacity {%s}",
+			cStorVolumeGot.Spec.Capacity,
+		)
+	}
+	curCapacity, ok := cStorVolumeGot.Status.Capacity[corev1.ResourceStorage]
+	glog.Infof("current capacity %v desiredCapcity %v", curCapacity, desiredCapacity)
+	if !ok || curCapacity.Cmp(desiredCapacity) != 0 {
+		resourceList := corev1.ResourceList{
+			corev1.ResourceName(corev1.ResourceStorage): desiredCapacity,
+		}
+		glog.Infof("[Debug] Updating current cpacity")
+		cStorVolumeGot.Status.Capacity = resourceList
+	}
```

Comment:
We are updating the size in status twice, once in resizeCStorVolume and once over here.

Reply:
Updated the code